week-1/Homework/mandatory/5-magic-8-ball.js

```diff
@@ -46,21 +46,161 @@ Very doubtful.
+                      break;
+                      case 2:
+                      return  "Without a doubt";
+                      break;
+                      case 3:
+                      return "Yes - definitely.";
+                      break;
+                      case 4:
+                      return "You may rely on it";
+                      break;
+                      case 5:
+                      return  "As I see it, yes";
+                      break;
+                      case 6:
+                      return "Most likely";
+                      break;
+                      case 7:
+                    return "Outlook good";
+                    break;
+                    case 8: 
+                    return "Yes", "Signs point to yes";
+                      break;
+                      case 9:
+
+                    return "Don't count on it";
+                    break;
+                    case 10:
+                    return "My reply is no";
+                    break;
+                    case 11:
+                    return "My sources say no";
+                    break;
+                    case 12:
+                    return "Outlook not so good";
+                    break;
+                    case 13:
+                    return "Very doubtful";
+                    break;
+                    case 14:
+                      return "Reply hazy, try again";
+                      break;
+                      case 15:
+                    return "Ask again later";
+                    break;
+                    case 16:
+                    return  "Better not tell you now" ;
+                    break;
+                    case 17:
+
+                    return "Cannot predict now";
+                    break;
+                    case 18:
+                    return"Concentrate and ask again" ;
+                    break;
+                    
+
+                   }
+                 
+                  
 }
+console.log(shakeBall());
+ 
+  ///This function check that outcome fit in to which criteria!!
+
+  function checkAnswer()
+  {
+    
+    var outcomes =  ["It is certain", 
+                   "It is decidedly so", 
+                   "Without a doubt", 
+                   "Yes - definitely",
+                   "You may rely on it", 
+                   "As I see it, yes", 
+                   "Most likely", 
+                   "Outlook good", 
+                   "Yes", "Signs point to yes",
+                   "Don't count on it", 
+                   "My reply is no",
+                   "My sources say no", 
+                   "Outlook not so good",
+                   "Very doubtful", 
+                   "Reply hazy, try again", 
+                   "Ask again later", 
+                   "Better not tell you now",
+                   "Cannot predict now", 
+                   "Concentrate and ask again"];
+
+    answer=Math.floor(Math.random()*outcomes.length);
```

Comment:
You are calculating the answer twice in two different parts of your code. You calculate the `answer` in `shakeBall` and then you calculate the `answer` again in `checkAnswer`. You don't need to do this calculation twice - how could you change this ?